backend/sftp/sftp.go

```diff
@@ -984,10 +984,10 @@ func NewFsWithConnection(ctx context.Context, f *Fs, name string, root string, m
 				outString := string(bytes.TrimSpace(stdout.Bytes()))
 				if strings.HasPrefix(outString, "Microsoft.PowerShell") { // If PowerShell: "Microsoft.PowerShell%ComSpec%"
 					f.shellType = "powershell"
-				} else if !strings.HasSuffix(outString, "%ComSpec%") { // If Command Prompt: "${ShellId}C:\WINDOWS\system32\cmd.exe"
+				} else if outString != "" && !strings.HasSuffix(outString, "%ComSpec%") { // If Command Prompt: "${ShellId}C:\WINDOWS\system32\cmd.exe"
```

Comment:
It would be better to detect this on a positive feature if at all possible, otherwise the next strange unix shell is going to be detected as a Windows cmd prompt.

How about containing case insensitive ".exe" or ".com" ?

Reply:
> It would be better to detect this on a positive feature if at all possible, otherwise the next strange unix shell is going to be detected as a Windows cmd prompt.

I see what you mean.

> How about containing case insensitive ".exe" or ".com" ?

Maybe, not sure, need to think about that.

Reply:
According to the Wikipedia article https://en.wikipedia.org/wiki/COMSPEC it should point to `COMMAND.COM`, `CMD.EXE` or `cmd.exe` - that was the basis of my suggestion. I don't think we'll ever get `.exe` or `.com` from anything other than Windows so it seemed like a nice positive ID to case insensitive search the string for `.exe` or `.com`.

Reply:
Sure, I agree it probably is the better solution after all. :)

Not very common, I think, but it can also point to tcc.exe, which is a compatible [alternative](https://jpsoft.com/products/tcc-le.html), and maybe there are others. I found an issue in the VSCode project requesting them to respect the ComSpec variable because the user used this TCC tool. And I also *think* it is possible to include arguments, like `cmd.exe /k`. And then I was thinking perhaps it is possible point ComSpec to some wrapper tool or script like `.vbs`, which in turn may launch cmd or tcc or something "behind the scenes", and then would we have to consider all of the extensions in PATHEXT (.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL)... 🤣  On the other hand the ComSpec variable is not very much used for anything in Windows itself anymore, I think, as there are other settings, e.g. in registry, to configure what is normally interpreted as the default shell, and ComSpec is now more like a de facto standard method for other programs to find the correct path to the shell, so I was starting to think it may not even be set at all... Aaaaaanyway, as you point out, it is probably better to get some strange cmd shell config to be auto-detected as Unix shell than the opposite - considering the amount sftp servers running the different shells, and also the amount of different shells which must be considered Unix shell with its escape rules. So... 😩 Will fix!

Reply:
Pushed a change. Did not think very much about performance, as it should normally run only once - except for ad-hoc remotes.. If you have further comments I'm happy to do another iteration before merging.